Sources/Plasma/NucleusLib/pnTimer/plTimerCallbackManager.cpp

```diff
@@ -122,16 +112,16 @@ bool plTimerCallbackManager::CancelCallbacksToKey(const plKey& key)
 {
     bool removed = false;
 
-    for (int i = fCallbacks.Count() - 1; i >= 0 ; i--)
+    for (auto iter = fCallbacks.rbegin(); iter != fCallbacks.rend(); ++iter)
     {
-        for (int j = 0; j < fCallbacks[i]->fMsg->GetNumReceivers(); j++)
+        for (size_t j = 0; j < (*iter)->fMsg->GetNumReceivers(); j++)
         {
-            const plKey rKey = fCallbacks[i]->fMsg->GetReceiver(j);
+            const plKey rKey = (*iter)->fMsg->GetReceiver(j);
             
             if (rKey == key)
             {
-                delete fCallbacks[i];
-                fCallbacks.Remove(i);
+                delete *iter;
+                fCallbacks.erase((iter + 1).base());    // Yuck
```

Comment:
Ok, I have to admit that I'm bamboozled here. I've been staring at this line for five minutes and still don't understand 😞

Reply:
Yes, this is an unfortunate ugly wart on STL...  You can't erase a `reverse_iterator`, so you have to convert it to a forward iterator...  But reverse iterators actually point to one after their "referenced" element (adjusted when dereferencing), so it needs to be moved forward one to get the correct element...   See the discussion [here](https://stackoverflow.com/questions/1830158/how-to-call-erase-with-a-reverse-iterator).

An alternate syntax would be `std::next(iter).base()`, but I'm not sure that's any more clear either.

Reply:
Ugh. Could the comment be amended to say something to the effect of `// Ugh - can't erase a reverse iterator`? I didn't realize that myself, and it would have (hopefully) been enough of a hint to go read up.

Maybe one day I will actually read the standard... 😛